pkg/kubectl/cmd/attach_test.go

```diff
@@ -86,6 +86,12 @@ func TestPodAndContainerAttach(t *testing.T) {
 			expectedContainer: "initfoo",
 			name:              "init container in flag",
 		},
+		{
+			p:           &AttachOptions{ContainerName: "wrong"},
```

Comment:
`TestPodAndContainerAttach` only checks arg setup. You'll want to add test cases in `TestAttach`, which uses a fake restclient.